### ISSUE TITLE
Rule to forbid(or warn about) disabling specific rules

### ISSUE THREAD

**Issue by toshi0383** [realm/SwiftLint#1308]
Sometimes people disable any rules because they just don't know how to fix it.

e.g.
```
// ("_" is a white space)
// swiftlint:disable trailing_whitespace
____
____// code_
____
```

I thought it would be useful if SwiftLint can detect when an "important rule" is disabled.

Basic idea is like this.

1. Detect disabled important rules in .swiftlint.yml

```.swiftlint.yml
important_rules:
  - trailing_whitespace
  - nested
disabled_rules:
  - trailing_whitespace
  - nested
```

Error Message:
```
'disabled_rules' cannot contain [trailing_whitespace, nested] since it's listed in 'important_rules'
```

2. Detect disabled important rules in each files

```.swiftlint.yml
important_rules:
  - trailing_whitespace
```

```a.swift
// swiftlint:disable:next trailing_whitespace
```

Error Message:
```
'trailing_whitespace' should not be disabled. (important_rules)
```

3. Ignore disable important_rules command
```swift
// swiftlint:disable important_rules <= ignored
```

If you guys feel happy with this idea, maybe I can work on a PullRequest.

**Comment by jpsim** [comment]
I'm not sure how I feel about this. I have a hard time thinking of _any_ rule in SwiftLint that should never be disabled...

**Comment by norio-nomura** [comment]
IMO, I feel it is too complicated. Can not meet your requirements with custom rules?
```yaml
custom_rules:
  force_unwrapping_is_important:
    name: "`force_unwrapping` should not be disabled."
    regex: "swiftlint:disable[^\n]+force_unwrapping"
    serverity: error
    match_kinds:
      - comment
```

**Comment by toshi0383** [comment]
Thanks for feedbacks!

@jpsim 
> I have a hard time thinking of any rule in SwiftLint that should never be disabled...

Lately I had to ask the other devs that "Please please follow the SwiftLint preference as much as you can, before simply disabling them." after noticing that they are suppressing a lot of rules blindly😕. I wanted that part to be handled by SwiftLint automatically.


@norio-nomura 
Yea I agree. Maybe this is too complicated idea.😅
But I'd say that important_rules looks simpler at least on `.swiftlint.yml` file than the custom_rule you suggest.
Of-course custom_rules can solve this issue, so it's perfectly okay to me, too.

Maybe we can consider implementing this again, when more users like me are 👍 about this?

**Comment by norio-nomura** [comment]
> But I'd say that important_rules looks simpler at least on .swiftlint.yml file than the custom_rule you suggest.

Indeed, you are right.🤔
I felt it might be good to have this function.

**Comment by marcelofabri** [comment]
My 2 cents: tooling can enforce/build team culture up to a point. There're ways to disable rules locally because exceptions do exist and a team should be able to recognize them and deal properly. 

Overall, I think it fits more as a custom rule than on SwiftLint itself.

**Comment by marcelofabri** [comment]
Closing this due to lack of activity.
